### ISSUE TITLE
Encode empty arrays

### ISSUE THREAD

**Issue by mitar** [sindresorhus/query-string#231]
For my use case it would be great if there would be a way to encode an empty array as well. For example, in `bracket` style, maybe `{a: []}` could be encoded as `a[]` only (while `{a: [1]}` is encoded as `a[]=1`).

**Comment by sindresorhus** [comment]
See my answer in https://github.com/sindresorhus/query-string/issues/228#issuecomment-575197530. It applies to this too.

**Comment by mitar** [comment]
Yes, but `bracket` style does not help here (which is your suggestion there).

So `bracket` style is less ambiguous, so adding `a[]` would solve this only ambiguous I know of which still exists for this style.

**Comment by sindresorhus** [comment]
That style is already used for null values: https://github.com/sindresorhus/query-string/blob/db66e68fc7e323995de713c38504a192472038fa/test/stringify.js#L117

**Comment by mitar** [comment]
I think `bar[]=` should be an empty array, while `bar` should be `null`, `bar=null` is string `null`, and `bar=` is empty string.

**Comment by sindresorhus** [comment]
Closing as implemented. Use `arrayFormat: 'bracket-separator'` (optionally with `arrayFormatSeparator`) to represent empty arrays.
> `stringify({a: []}) → 'a[]'` and `parse('a[]') → {a: []}`.
> `a[]=` remains an array with a single empty string, so the cases are unambiguous and round-trip.

**Comment by mitar** [comment]
Awesome, thanks!
